### ISSUE TITLE
Efficient Incremental Indexing for Large Repositories Using zoekt-git-index

### ISSUE THREAD

**Issue by bonohubby** [sourcegraph/zoekt#791]
Thank you for providing such an excellent tool.

I am planning to configure a Code Search service using zoekt-git-index.

I am curious if, after the initial setup, any changes in the repository and to provide up-to-date search, it would require re-indexing the entire content with zoekt-git-index to create a new `.zoekt` file? Or is it possible to reflect the changes using the existing `.zoekt` file by only indexing the modified content?

I thought that this functionality would be necessary for large repositories and I would like to know whether zoekt has this feature.

I would greatly appreciate it if you could provide an answer.

**Comment by keegancsmith** [comment]
Hello. We have found even with repos that have multiple gigabyte working copies it isn't such a big deal to re-index and at Sourcegraph that is our default behaviour even for large monorepos. However, we do have functionality called "delta indexing" which will do what you ask. We have validated it works, but due to some changes in focus we never shipped it as the default. I recommend playing around with the `-delta` and `-delta_threshold` flags. We default to a value of 150 for delta_threshold. It is essentially the number of incremental re-indexes we do until we redo a full index. We have other functionality called shard merging which would avoid that need, but we never integrated to two.
